x-pack/plugins/ml/public/application/jobs/jobs_list/components/utils.js

```diff
@@ -76,6 +77,10 @@ export function isClosable(jobs) {
   );
 }
 
+export function isResetable(jobs) {
+  return jobs.some((j) => j.jobState === JOB_STATE.CLOSED);
```

Comment:
Should the reset action be shown for jobs that have never run (or ones which have already been reset)?

Reply:
I don't think there is any harm in showing the reset action for jobs which haven't been run or have already been reset.
How would we determine that? `closed` and no processed records?

This is a similar question to whether we should allow a job to be reset again while a ongoing reset task is running.
For deleting we allow the user to click delete again if the job is "deleting", i think in case the delete task is hanging.
@dimitris-athanasiou what do you think?
Also, should we allow a job to be deleted if an ongoing reset or revert task is running? or should we only allow the user to re-click the current task type?

Reply:
With regard to resetting a job that has never run or that has just been reset: there is no harm indeed. I would argue to keep the logic simple and let the reset action be shown in those cases. If the user hits the reset button again, it will almost be a no-op.

> Also, should we allow a job to be deleted if an ongoing reset or revert task is running? or should we only allow the user to re-click the current task type?

A job can be deleted while a reset or revert task is running.

But if a reset or revert task is running, the job should not be allowed to be opened nor the other blocking action should be allowed (revert when reset is running and vice versa). If the user calls, for example, revert while reset is happening, they will get an error message letting them know the job is blocked while resetting. The UI would ideally disable the relevant action with an informative tooltip.

Reply:
> But if a reset or revert task is running, the job should not be allowed to be opened nor the other blocking action should be allowed 

Yes, we disable all other options, other than delete. but isn't delete a blocking action?

Is there any benefit in allowing the user to reselect `reset` if a reset task is already running? Like we do with delete.

So just to be clear.. when a delete, revert or reset action is running, the only action that should be available is delete?